File: dotnet/cafdemalihapa/Directories/Hosting/Feature/BackEnd/BackEndDirectory.cs
namespace cafdemalihapa.Directories.Hosting.Feature.BackEnd
{
    public class BackEndDirectory
    (
        FeatureDirectory featureDirectory
    )
    {
        public string GetName()
        {
            return "bend";
        }
        public string GetPath()
        {
            var directoryName = GetName();
            var featureDirectoryPath = featureDirectory.GetPath();
            var notesAndMessages = Path.Combine(featureDirectoryPath, directoryName);
            return notesAndMessages;
        }
        public void Create()
        {
            var path = GetPath();
            if(!Directory.Exists(path))
            {
                Directory.CreateDirectory(path);
            }
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/BackEnd/BackEndDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.BackEnd

open System.IO
open cafdemwimu.console.Directories.Hosting.Feature

[<AbstractClass; Sealed>]
type BackEndDirectory private () =
    static member GetName() = "bend"

    static member GetPath() =
        let directoryName = BackEndDirectory.GetName()
        let featureDirectoryPath = FeatureDirectory.GetPath()
        let notesAndMessages = Path.Combine(featureDirectoryPath, directoryName)
        notesAndMessages

    static member Create() =
        let path = BackEndDirectory.GetPath()
        if not (Directory.Exists(path)) then
            Directory.CreateDirectory(path) |> ignore
